ql/src/java/org/apache/hadoop/hive/ql/io/orc/VectorizedOrcAcidRowBatchReader.java

```diff
@@ -892,13 +913,20 @@ public boolean next(NullWritable key, VectorizedRowBatch value) throws IOExcepti
     } catch (Exception e) {
       throw new IOException("error iterating", e);
     }
-    if(!includeAcidColumns) {
+    if (!includeAcidColumns) {
       //if here, we don't need to filter anything wrt acid metadata columns
       //in fact, they are not even read from file/llap
       value.size = vectorizedRowBatchBase.size;
       value.selected = vectorizedRowBatchBase.selected;
       value.selectedInUse = vectorizedRowBatchBase.selectedInUse;
       copyFromBase(value);
+
+      if (rowIsDeletedProjected) {
+        rowIsDeletedVector.clear();
+        int ix = rbCtx.findVirtualColumnNum(VirtualColumn.ROWISDELETED);
```

Comment:
Why do we have to recompute this for every batch? Lets store this along with rowIsDeletedProjected flag

Reply:
I started to work on a solution to manage Virtual Column related information but it lead to a much bigger change. `VectorizedOrcAcidRowBatchReader` can behave several ways and each of those behavior worth a separate class after extracting common parts.
So I decided to followed existing logic implemented for RowId.